### PR TITLE
chore: switch to pinned nanoversion strategy

### PR DESCRIPTION
**Pull request by vvcephei**
Switches the build to pinned nanoversions.

Depends on https://github.com/confluentinc/jenkins-common/pull/474 and https://github.com/confluentinc/ci-tools/pull/101

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -335,25 +333,25 @@
...
                 <artifactId>kafka-connect-json-schema-converter</artifactId>
-                <version>${io.confluent.schema-registry.version}</version>
+                <version>7.3.0-615</version>
             </dependency>
 
             <dependency>
                 <groupId>io.confluent</groupId>
                 <artifactId>kafka-connect-protobuf-converter</artifactId>
-                <version>${io.confluent.schema-registry.version}</version>
+                <version>7.3.0-615</version>
```
Comment (jnh5y) [inline pom.xml:354]: Just to confirm, the common pom version and the kafka.version are different, right?

**Review by jnh5y — APPROVED** [review]
I'm excited to try out the pinned versions!

**Review by guozhangwang — APPROVED** [review]
+1, excited to see this happening, finally!

File: pom.xml (hunk 1 above)
Comment (vvcephei) [reply pom.xml:354]: Yes. The `kafka.version` is actually set in the common pom. This line is the schema registry version, though, and it's also different than the common pom.

**Comment by vvcephei** [comment]
The last build failed because the agent ran out of disk space, but a prior one passed, so I'll go ahead and merge.

**Pull request merged**